src/components/delete-user-modal.jsx

```diff
@@ -0,0 +1,54 @@
+import { useMutation } from "@apollo/client";
+import { useContext } from "react";
+import { DELETE_USER } from "../graphql/delete-user";
+import { CurrentUserContext } from "../store/current-user-context";
+import { Button } from "./button";
+import { Modal } from "./modal";
+
+export function DeleteUserModal({ onClose }) {
+  const token = localStorage.getItem("jwt");
+  const [deleteUser, { data, loading, error }] = useMutation(DELETE_USER);
+  const { currentUser } = useContext(CurrentUserContext);
+  const userId = currentUser.id;
+
+  const handleCloseModal = () => {
+    onClose();
+  };
+
+  const handleDeleteUserSubmit = () => {
+    //loading visible
+    //call deleteUser(userID)
+    deleteUser({
+      variables: {
+        where: { id: userId },
+      },
+    })
+      //.then(res)
+      .then(({ data }) => {
+        //console.log(data);
+        // - - - delete stored token
+        localStorage.removeItem("jwt");
+        // - - - close modal
+        handleCloseModal();
+        window.location.reload();
+        // - - - remove "loading"
+        // - - - redirect to home page
+        // - - - or run "logout" when that is setup
+      })
+      //.catch
+      .catch(err);
+    //.finally?
+  };
+
+  return (
+    <Modal title="Are you sure?">
```

Comment:
@dani9401 I created a delete-confirmation-modal.jsx that you can reuse and it is already under main. It looks super similar to this but might be worth reusing a component instead of creating it from scratch. Should be an easy switch over too. Thoughts on this? @SamPage4427